src/inspect.rs

```diff
@@ -1,24 +1,202 @@
 use anyhow::Result;
 use biscuit_auth::{
-    builder::{Fact, Policy, Rule},
+    builder::{Fact, Rule},
     datalog::RunLimits,
     error::{FailedCheck, Logic, MatchedPolicy, RunLimit, Token},
     Authorizer, UnverifiedBiscuit,
 };
 use chrono::offset::Utc;
+use serde::Serialize;
 use std::fs;
 use std::path::PathBuf;
 
 use crate::cli::*;
 use crate::errors::CliError::*;
 use crate::input::*;
 
+#[derive(Serialize)]
+struct TokenBlock {
+    code: String,
+    external_key: Option<String>,
+    revocation_id: String,
+}
+
+#[derive(Serialize)]
+struct TokenDescription {
+    sealed: bool,
+    root_key_id: Option<u32>,
+    blocks: Vec<TokenBlock>,
+}
+
+impl TokenDescription {
+    fn render(&self) {
```

Comment:
another option should be to implement a Serde text de-serializer for `TokenDescription`

Reply:
do you mean serializer?

I don't think it's worth the complexity (and i'm not even sure it would be doable)

Reply:
do you mean serializer?

I don't think it's worth the complexity (and i'm not even sure it would be doable)

Reply:
Yes, you’re right , a serializer.
I agree it’s a little more work (and should be managed in another PR) but allow more homogeneous code.

Reply:
At an `Into` impl could do the job : 

```
impl From<TokenDescription> for String {
    fn from(val: TokenDescription) -> Self {
        let mut builder = Builder::default();
        match val.sealed {
            true => {
                builder.append("Sealed biscuit\n");
            }
            false => {
                builder.append("Open biscuit\n");
            }
        }

        val.blocks.iter().enumerate().for_each(|(i, block)| {
            match i == 0 {
                true => match val.root_key_id {
                    Some(root_key_id) => {
                        builder.append(format!(
                            "Authority block (root key identifier: {:?}):\n",
                            &root_key_id
                        ));
                    }
                    None => {
                        builder.append("Authority block:\n");
                    }
                },
                false => match &block.external_key {
                    Some(epk) => {
                        builder.append(format!(
                            "Block n°{}, (third party, signed by {}):\n",
                            i, epk
                        ));
                    }
                    _ => {
                        builder.append(format!("Block n°{}:\n", i));
                    }
                },
            }

            builder.append("== Datalog ==\n");
            builder.append(block.code.clone());

            builder.append("== Revocation id ==\n");
            builder.append(block.revocation_id.clone());
            builder.append("\n==========\n");
        });
        match builder.string() {
            Ok(str) => str,
            Err(e) => {
                format!("unable to render token : {}\n", e)
            }
        }
    }
}
```

Reply:
i was thinking of something along the lines of `Display`, a function taking a buffer and writing to it. we could then use `to_string()`